scala-generator/src/main/scala/models/http4s/CirceJson.scala

```diff
@@ -46,32 +46,38 @@ ${Seq(generateTimeSerde(), generateEnums(), generateModels(), generateUnions()).
   }
 
   def generateModels(): String = {
-    Seq(
-      ssd.models.map(decodersAndEncoders(_)).mkString("\n\n"),
-      PrimitiveWrapper(ssd).wrappers.map(w => decodersAndEncoders(w.model)).mkString("\n\n")
-    ).filter(!_.trim.isEmpty).mkString("\n\n")
+    (ssd.models ++ PrimitiveWrapper(ssd).wrappers.map(_.model))
+      .flatMap(decodersAndEncoders)
+      .filter(_.trim.nonEmpty)
+      .mkString("\n\n")
   }
 
   def generateUnions(): String = {
-    ssd.unions.map(decodersAndEncoders(_)).mkString("\n\n")
+    ssd.unions.map(decodersAndEncoders).mkString("\n\n")
   }
 
   /**
     * Returns the implicits for enum json serialization, handling
     * conversion both from the string and object representations.
     */
   def generateEnums(): String = {
-    ssd.enums.map(enumDecodersAndEncoders(_)).mkString("\n\n")
-  }
-
-  private[models] def enumDecodersAndEncoders(enum: ScalaEnum): String = {
-    Seq(
-      s"""implicit val jsonDecoder${ssd.name}${enum.name}: Decoder[${enum.qualifiedName}] =""",
-      s"""  Decoder.decodeString.map(${enum.qualifiedName}(_))""",
-      "",
-      s"""implicit val jsonEncoder${ssd.name}${enum.name}: Encoder[${enum.qualifiedName}] =""",
-      s"""  Encoder.encodeString.contramap[${enum.qualifiedName}](_.toString)"""
-    ).mkString("\n")
+    ssd.enums.flatMap(enumDecodersAndEncoders).mkString("\n\n")
+  }
+
+  private[this] def enumDecodersAndEncoders(enum: ScalaEnum): Option[String] = {
+    if (enum.values.isEmpty) {
+      None
+    } else {
+      Some(
```

Comment:
Instead of `if (...) None else Some(...)` could do `enum.values.headOption.map { _ => ...}`